syntax = "proto3";                     // 指定 proto 语法版本为 proto3
import "base.proto";                // 导入公共基础消息定义

// ========================== 背包物品 ==========================
message BagItem {
    uint64   guid         = 1;          // 唯一标识符（物品在背包中的唯一ID）
    uint64   item_guid    = 2;          // 道具模板Guid（区分同类不同实例）
    int32    item_id      = 3;          // 道具ID（静态表配置的编号）
    int64    item_num     = 4;          // 道具数量
    bool     bind         = 5;          // 是否绑定（true=绑定，false=非绑定）
}

message BagChangeNty {
    repeated BagItem change_list = 1;   // 变更的物品列表（新增或修改）
    repeated uint64  remove_list = 2;   // 被移除的物品ID列表
}

// ========================== 装备物品 ==========================
message EquipItem {
    uint64   guid           = 1;        // 唯一标识符
    int32    equip_id       = 2;        // 装备ID
    int32    strength_lvl   = 3;        // 强化等级
    int32    refine_level   = 4;        // 精炼等级
    int32    star_level     = 5;        // 星级
    int32    refine_exp     = 6;        // 精炼经验
    int32    star_exp       = 7;        // 星级经验
    int32    is_using       = 8;        // 是否上阵（0=否，1=是）
}

message EquipChangeNty {
    repeated EquipItem change_list = 1; // 装备变化列表
    repeated uint64    remove_list = 2; // 移除的装备ID
}

// ========================== 宝石物品 ==========================
message GemItem {
    uint64   guid           = 1;        // 唯一标识符
    int32    gem_id         = 2;        // 宝石ID
    int32    strength_lvl   = 3;        // 强化等级
    int32    refine_level   = 4;        // 精炼等级
    int32    star_level     = 5;        // 星级
    int32    refine_exp     = 6;        // 精炼经验
    int32    star_exp       = 7;        // 星级经验
    int32    pos            = 8;        // 镶嵌位置
}

message GemChangeNty {
    repeated GemItem change_list = 1;   // 宝石变化列表
    repeated uint64  remove_list = 2;   // 移除的宝石ID
}

// ========================== 宠物物品 ==========================
message PetItem {
    uint64 guid          = 1;           // 唯一标识符
    int32  pet_id        = 2;           // 宠物ID
    int32  strength_lvl  = 3;           // 强化等级
    int32  refine_level  = 4;           // 精炼等级
    int32  star_level    = 5;           // 星级
    int32  refine_exp    = 6;           // 精炼经验
    int32  star_exp      = 7;           // 星级经验
    int32  is_using      = 8;           // 是否上阵
}

message PetChangeNty {
    repeated PetItem change_list = 1;   // 宠物变化列表
    repeated uint64  remove_list = 2;   // 移除的宠物ID
}

// ========================== 伙伴物品 ==========================
message PartnerItem {
    uint64 guid          = 1;           // 唯一标识符
    int32  partner_id    = 2;           // 伙伴ID
    int32  strength_lvl  = 3;           // 强化等级
    int32  refine_level  = 4;           // 精炼等级
    int32  star_level    = 5;           // 星级
    int32  refine_exp    = 6;           // 精炼经验
    int32  star_exp      = 7;           // 星级经验
    int32  set_pos       = 8;           // 上阵位置
}

message PartnerChangeNty {
    repeated PartnerItem change_list = 1; // 伙伴变化列表
    repeated uint64      remove_list = 2; // 移除的伙伴ID
}

// ========================== 坐骑物品 ==========================
message MountItem {
    uint64 guid          = 1;           // 唯一标识符
    int32  mount_id      = 2;           // 坐骑ID
    int32  strength_lvl  = 3;           // 强化等级
    int32  refine_level  = 4;           // 精炼等级
    int32  star_level    = 5;           // 星级
    int32  refine_exp    = 6;           // 精炼经验
    int32  star_exp      = 7;           // 星级经验
    int32  is_using      = 8;           // 是否出战
}

message MountChangeNty {
    repeated MountItem change_list = 1; // 坐骑变化列表
    repeated uint64    remove_list = 2; // 移除的坐骑ID
}

// ========================== 邮件 ==========================
message MailItem {
    uint64 guid           = 1;          // 唯一标识符
    int32  mail_type      = 2;          // 邮件类型
    int32  status         = 3;          // 邮件状态（0=未读，1=已读等）
    string title          = 4;          // 邮件标题
    string content        = 5;          // 邮件内容
    string sender         = 6;          // 发件人
    repeated int32 item_id  = 8;        // 附件物品ID列表
    repeated int32 item_num = 9;        // 附件物品数量列表
}

message MailChangeNty {
    repeated MailItem change_list = 1;  // 邮件变化列表
    repeated uint64   remove_list = 2;  // 移除的邮件ID
}

// ========================== 任务 ==========================
message TaskItem {
    uint32  task_id = 1;                // 任务ID
}

message TaskChangeNty {
    repeated TaskItem change_list = 1;  // 任务变化列表
    repeated uint64   remove_list = 2;  // 移除的任务ID
}

// ========================== 副本与章节 ==========================
message CopyItem {
    uint32  copy_id    = 1;             // 副本ID
    int32   star_num   = 2;             // 获得的星数
    int32   battle_cnt = 3;             // 挑战次数
    int32   reset_cnt  = 4;             // 重置次数
}

message ChapterItem {
    uint32  chapter_id  = 1;            // 章节ID
    uint32  star_award  = 2;            // 星数奖励领取状态
    uint32  scene_award = 3;            // 场景奖励领取状态
    uint32  copy_type   = 4;            // 副本类型
}

// ========================== 行为记录 ==========================
message ActionItem {
    int64 action = 1;                   // 行为类型
    int64 actime = 2;                   // 行为发生时间戳
}

// ========================== 引导项 ==========================
message GuideItem {
    int32 id = 1;                       // 引导步骤ID
}

// ========================== 技能变化 ==========================
message SkillChangeNty {
    repeated SkillItem change_list = 1; // 技能变化列表
    repeated uint64    remove_list = 2; // 移除的技能ID
}

// ========================== 登录返回 ==========================
message RoleLoginAck {
    uint32   ret_code       = 1;        // 返回码（0=成功，非0=错误码）
    uint64   account_id     = 2;        // 账号ID
    uint64   role_id        = 3;        // 角色ID
    int32    carrer         = 4;        // 职业
    uint32    level          = 5;        // 等级
    int64    exp            = 6;        // 经验
    string   name           = 7;        // 角色名
    int32    vip_lvl        = 8;        // VIP等级
    int32    vip_exp        = 9;        // VIP经验
    int64    fight_value    = 10;       // 战斗力
    repeated ActionItem  action_list   = 11; // 行为记录
    repeated EquipItem   equip_list    = 12; // 装备列表
    repeated GemItem     gem_list      = 13; // 宝石列表
    repeated PetItem     pet_list      = 14; // 宠物列表
    repeated PartnerItem partner_list  = 15; // 伙伴列表
    repeated MountItem   mount_list    = 16; // 坐骑列表
    repeated BagItem     bag_item_list = 17; // 背包物品列表
    repeated MailItem    mail_list     = 18; // 邮件列表
    repeated TaskItem    task_list     = 19; // 任务列表
    repeated CopyItem    copy_list     = 20; // 副本列表
    repeated ChapterItem chapter_list  = 21; // 章节列表
    repeated SkillItem   skill_list    = 22; // 技能列表
}
